nautobot/core/tests/runner.py

```diff
@@ -55,7 +57,25 @@ def setup_databases(self, **kwargs):
         result = super().setup_databases(**kwargs)
 
         if settings.TEST_USE_FACTORIES and result:
-            command = ["generate_test_data", "--flush", "--no-input"]
+            # Try to flush the database before loading test data.
+            # Experienced fails when flushing multiple test databases concurently, so we retry a few times.
+            for connection in result:
+                db_name = connection[0].alias
+
+                retry_count = 4
+                retry_wait = 1
+                while retry_count:
+                    try:
+                        print(f'Flushing test database "{db_name}"...')
+                        call_command("flush", "--no-input", "--database", db_name)
+                        break
+                    except CommandError as e:
+                        retry_count -= 1
+                        sleep(retry_wait)
+                        retry_wait *= 2
+                        print(f"Flush command failed with error: {e}, will retry {retry_count} more times.")
+
```

Comment:
Don't we need an `else` on the `while` loop to fail the test as a whole if we retry `retry_count` times and still fail every time?

Reply:
Fixed